src/api/api.ts

```diff
@@ -0,0 +1,52 @@
+import { Body, fetch as tauriFetch, HttpVerb } from '@tauri-apps/api/http';
+
+import { REFSTUDIO_HOST } from './server';
+
+/** Issue a GET request using either web fetch or Tauri fetch */
+export async function universalGet<ResponsePayload = unknown>(path: string): Promise<ResponsePayload> {
+  if (import.meta.env.VITE_IS_WEB) {
+    const response = await fetch(path);
+    if (!response.ok) {
+      throw new Error(`Failed to get ${path}: ${response.statusText}`);
+    }
+    return response.json() as ResponsePayload;
+  } else {
+    const response = await tauriFetch(REFSTUDIO_HOST + path);
+    if (!response.ok) {
+      throw new Error(`Failed to get ${path}: ${response.status}`);
+    }
+    return response.data as ResponsePayload;
+  }
+}
+
+/** Issue a POST request with a JSON payload using either web fetch or Tauri fetch */
+export async function universalPost<ResponsePayload = unknown, RequestPayload = unknown>(
+  path: string,
+  payload: RequestPayload,
+  method: HttpVerb = 'POST',
```

Comment:
nit: we should not be able to pass a method, or we should rename the function to `universalFetch`

Reply:
Ok. Will do that. Agree.